kpi/tests/api/v2/test_api_submissions.py

```diff
@@ -233,6 +238,53 @@ def test_list_submissions_authenticated_asset_publicly_shared(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.asset.remove_perm(anonymous_user, PERM_VIEW_SUBMISSIONS)
 
+    def test_list_submissions_asset_publicly_shared_and_shared_with_user(self):
+        """
+        Running through behaviour described in issue kpi/#2870 where an asset
+        that has been publicly shared and then explicity shared with a user, the
+        user has lower permissions than an anonymous user and is therefore
+        unable to view submission data.
+        """
+
+        self._log_in_as_another_user()
+        anonymous_user = get_anonymous_user()
+
+        assert self.asset.has_perm(self.anotheruser, PERM_VIEW_ASSET) == False
+        assert PERM_VIEW_ASSET not in self.asset.get_perms(self.anotheruser)
+        assert self.asset.has_perm(self.anotheruser, PERM_CHANGE_ASSET) == False
+        assert PERM_CHANGE_ASSET not in self.asset.get_perms(self.anotheruser)
+
+        self.asset.assign_perm(self.anotheruser, PERM_CHANGE_ASSET)
```

Comment:
Only `'view_submissions'`  is needed to export data. User don't need `'change_asset'`. You can remove it and related assertions to this permission.